### PR TITLE
feat: [P4ADEV-207] get transaction receipt

### PR DESCRIPTION
**Pull request by mmoio**
#### List of Changes

- GET transactions/receipt{transactionId} in transactionDetail component
- Download button is disabled while request is pending or on error
- useReceiptData hook for api call and normalization
- unit tests

#### Motivation and Context

Initial implementation of the get transaction receipt, this pr is focused mainly on the success case with minimal error handling. For this implementation and for testing purposes the [arc-be-mock](https://github.com/pagopa/pagopa-arc-be-mock) and its openapi spec have been used as reference.
#### How Has This Been Tested?
- Visual testing with [arc-be-mock](https://github.com/pagopa/pagopa-arc-be-mock) on branch `develop`
- Unit tests

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -9,7 +9,7 @@ import { theme } from '@pagopa/mui-italia';
 import UserRoute from 'routes/User';
 import { RouteHandleObject } from 'models/Breadcrumbs';
 import TransactionsList from 'routes/TransactionsList';
-import utils from 'utils';
+import utils from './utils';
```
Comment (stratoivandiluccio) [inline src/App.tsx:12]: From 'utils' should be ok. This is becuase in tsconfig.json you cna find:

`"baseUrl": "src",`

Comment (mmoio) [reply src/App.tsx]: Ok, I've amended my last commit

File: src/components/Transactions/transactionDetail.test.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      i18n: {
+        changeLanguage: () => new Promise(() => {})
+      }
+    };
+  },
+  initReactI18next: {
+    type: '3rdParty',
+    init: () => {}
+  }
+}));
```
Comment (stratoivandiluccio) [inline src/components/Transactions/transactionDetail.test.tsx:22]: Can we make this mock available by default in every test?

Should be possibile moving this code in another file and referencing it by the `setupFile` property in `jest.config.js` file.

https://stackoverflow.com/questions/49685265/jest-mock-user-module-in-all-test-files

**Comment by stratoivandiluccio** [comment]
Great work here!

**Review by stratoivandiluccio — APPROVED** [review]

**Pull request merged**